webapp/src/types/api.d.ts

```diff
@@ -12,6 +12,12 @@ export interface APIError {
   message: string;
 }
 
+export interface Competency {
```

Comment:
I'm curious if this needs to have `extends Entity` after Competency? That's what it shows on similar entries like Reflection.

Reply:
I believe that it should.

Reply:
In that case, `Doc` below is also non-conforming. Perhaps this is a good chance to address both of these issues to extend `Entity` and remove the `id` property.